File: misc/benchmarks/reference/java/InlineStress.java
/** InlineStress.java — Inlining stress benchmark (50M iterations).
    Equivalent to examples/il/benchmarks/inline_stress.il */
public class InlineStress {
    static long doubleVal(long x) {
        return x + x;
    }

    static long square(long x) {
        return x * x;
    }

    static long add3(long a, long b, long c) {
        return a + b + c;
    }

    static long inc(long x) {
        return x + 1;
    }

    static long combine(long x) {
        long d = doubleVal(x);
        long s = square(x);
        long i = inc(x);
        return add3(d, s, i);
    }

    public static void main(String[] args) {
        long n = 50000000 + args.length;
        long sum = 0;
        for (long i = 0; i < n; i++) {
            long r = combine(i);
            long rawSum = sum + r;
            sum = rawSum & 268435455;
        }
        System.exit((int)(sum & 0xFF));
    }
}

File: misc/benchmarks/reference/python/inline_stress.py
"""inline_stress.py — Inlining stress benchmark (50M iterations).
Equivalent to examples/il/benchmarks/inline_stress.il"""
import sys

def double_val(x):
    return x + x

def square(x):
    return x * x

def add3(a, b, c):
    return a + b + c

def inc(x):
    return x + 1

def combine(x):
    d = double_val(x)
    s = square(x)
    i = inc(x)
    return add3(d, s, i)

s = 0
for i in range(50000000):
    r = combine(i)
    raw_sum = s + r
    s = raw_sum & 268435455

sys.exit(s & 0xFF)
